data/mods/blazemod/blaze_ammo.json

```diff
@@ -1,412 +1,386 @@
+  },
+  {
+    "id": "h_projectile",
+    "type": "AMMO",
+    "name": "modified mininuke",
+    "description": "A heavy modified handheld nuclear device.  Intended to be fired from a specialized launcher, its case has been cut down and it has been rigged to explode on impact.  While it is now somewhat lighter than before, it can no longer be triggered manually.",
+    "weight": 3816,
+    "volume": 1,
+    "price": 180000,
+    "bashing": 8,
+    "material": "plastic",
+    "symbol": "*",
+    "color": "green",
+    "count": 1,
+    "stack_size": 2,
+    "ammo_type": "h_projectile",
+    "damage": 14,
+    "effects": [ "MININUKE_MOD", "NEVER_MISFIRES" ],
+    "flags": [ "LEAK_RAD" ]
+  },
+  {
+    "id": "harpoon",
+    "type": "AMMO",
+    "name": "harpoon",
+    "description": "A large spear carved from wood.  This was fashioned specifically to be fired from a compatible weapon; thus unsuitable for melee combat.",
+    "weight": 1455,
+    "volume": 10,
+    "price": 1000,
+    "bashing": 5,
+    "material": "wood",
+    "symbol": "=",
+    "color": "brown",
+    "count": 10,
+    "ammo_type": "harpoon",
+    "damage": 40,
+    "pierce": 10,
+    "effects": [ "RECOVER_3", "NEVER_MISFIRES" ]
+  },
+  {
+    "id": "hbolt_boom",
+    "type": "AMMO",
+    "name": "nova bolt",
+    "description": "An IED composed of a sealed container filled with a payload material affixed to the end of a large ballista bolt.  Intended to be fired from a specialized launcher.  Carries a substantial explosive payload.",
+    "weight": 3816,
+    "volume": 1,
+    "price": 1000,
+    "bashing": 8,
+    "material": "iron",
+    "symbol": "*",
+    "color": "red",
+    "count": 1,
+    "stack_size": 1,
+    "ammo_type": "hbolt",
+    "damage": 14,
+    "effects": [ "NEVER_MISFIRES", "EXPLOSIVE_BIG", "COOKOFF" ],
+    "flags": [ "" ]
+  },
+  {
+    "id": "hbolt_boom2",
+    "type": "AMMO",
+    "name": "supernova bolt",
+    "description": "An IED composed of a sealed container filled with a payload material affixed to the end of a large ballista bolt.  Intended to be fired from a specialized launcher.  Carries a very substantial explosive payload.",
+    "weight": 3816,
+    "volume": 1,
+    "price": 1000,
+    "bashing": 8,
+    "material": "iron",
+    "symbol": "*",
+    "color": "red",
+    "count": 1,
+    "stack_size": 1,
+    "ammo_type": "hbolt",
+    "damage": 14,
+    "effects": [ "NEVER_MISFIRES", "EXPLOSIVE_HUGE", "COOKOFF" ],
+    "flags": [ "" ]
+  },
+  {
+    "id": "hbolt_fire",
+    "type": "AMMO",
+    "name": "searing bolt",
+    "description": "An IED composed of a container filled with a payload material affixed to the end of a large ballista bolt.  Intended to be fired from a specialized launcher.  This one is designed to douse a small area with deadly flames.",
+    "weight": 3816,
+    "volume": 1,
+    "price": 600,
+    "bashing": 8,
+    "material": "iron",
+    "symbol": "*",
+    "color": "red",
+    "count": 1,
+    "stack_size": 1,
```

Comment:
Yes, but that is out of scope of the linter - it's valid JSON, a valid field and sensibly formatted.

Strict mode (#17722 etc) could detect this but cannot be enabled by default for mods due to the false positives that would be reported due to the variation in load order. One option would be a command-line switch that selectively tested a mod in strict mode and output a non-compliance log?